Plots/Meteograms/NCL_meteo_1.py

```diff
@@ -0,0 +1,276 @@
+"""
+NCL_meteo_1.py
+===============
+
+This script illustrates the following concepts:
+   - Drawing a meteogram
+   - Creating a color map using hex values
+   - Reversing the Y axis
+   - Explicitly setting tickmarks and labels on the bottom X axis
+   - Increasing the thickness of contour lines
+   - Drawing wind barbs
+   - Drawing a bar chart
+   - Changing the width and height of a plot
+   - Overlaying wind barbs and line contours on filled contours
+   - Changing the position of individual plots on a page
+
+See following URLs to see the reproduced NCL plot & script:
+    - Original NCL script: https://www.ncl.ucar.edu/Applications/Scripts/meteo_1.ncl
+    - Original NCL plot: https://www.ncl.ucar.edu/Applications/Images/meteo_1_lg.png
+"""
+
+###############################################################################
+# Import necessary packages
+import xarray as xr
+import numpy as np
+from matplotlib import pyplot as plt
+from matplotlib.colors import ListedColormap, BoundaryNorm
+import cartopy.crs as ccrs
+
+import geocat.datafiles as gdf
+from geocat.viz import util as gvutil
+
+
+###############################################################################
+# Read in data:
+
+# Open a netCDF data file using xarray default engine
+# and load the data into xarray
+ds = xr.open_dataset(gdf.get("netcdf_files/meteo_data.nc"), decode_times=False)
+
+# Extract variables from the data
+tempisobar = ds.tempisobar
+levels = ds.levels
+taus = ds.taus
+rh = ds.rh
+ugrid = ds.ugrid
+vgrid = ds.vgrid
+rain03 = ds.rain03
+tempht = ds.tempht
+
+###############################################################################
+# Plot:
+
+# Generate figure (set its size (width, height) in inches)
+fig = plt.figure(figsize=(6, 8))
+spec = fig.add_gridspec(ncols=1, nrows=3, height_ratios=[4, 1, 1], hspace=0.4)
+
+# Create axis for contour/wind barb plot
+ax1 = fig.add_subplot(spec[0, 0], projection=ccrs.PlateCarree())
+
+# Add coastlines to first axis
+ax1.coastlines(linewidths=0.5, alpha=-1)
+
+# Set aspect ratio of the first axis
+ax1.set_aspect(2)
+
+# Create a color map with a combination of matplotlib colors and hex values
+colors = ListedColormap(np.array(['white', 'white', 'white', 'white', 'white',
+                                  'mintcream', "#DAF6D3", "#B2FAB9", "#B2FAB9",
+                                  'springgreen', 'lime', "#54A63F"]))
+bounds = [-20, -10, 0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100]
```

Comment:
"contour_levels", or even "levels" would be more a more meaningful variable name than "bounds". Ditto "normalized_levels" would be better than "norm"?

Reply:
Ok, I made this change.